Chat/assignment-chat/client-chat/src/main/java/ru/assignment/net/Receiver.java

```diff
@@ -1,47 +1,112 @@
 package ru.assignment.net;
 
+import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.net.Socket;
-import java.util.Scanner;
+import java.nio.charset.StandardCharsets;
 
 /**
  * Created by Андрей on 20.02.2015.
  */
 public class Receiver implements Runnable {
-    private Socket clientSocket;
-    private Sender sender;
-    private Scanner reader;
+    private final Socket clientSocket;
+    private final BufferedReader reader;
+    private final InputStreamReader streamReader;
+    private final DisconnectReceivedListener listener;
+    private volatile boolean isOpen = false;
+    private Thread currentThread;
 
-    public Receiver(Socket clientSocket, Sender sender) throws IOException {
+
+    public Receiver(Socket clientSocket, DisconnectReceivedListener listener) throws IOException {
+        ChatClient.clientLogger.debug("Configuration receiver constructor");
+        this.listener = listener;
         this.clientSocket = clientSocket;
-        reader = new Scanner(clientSocket.getInputStream());
-        this.sender = sender;
+        streamReader = new InputStreamReader(clientSocket.getInputStream(), StandardCharsets.UTF_16);
+        reader = new BufferedReader(streamReader);
+    }
+
+    public void connectAsync() {
+        ChatClient.clientLogger.debug("Start new receiver thread");
+        Thread receiverThread = new Thread(this);
+        receiverThread.start();
     }
 
     public void run() {
-        System.out.println("run receiver");
-        listen();
+        ChatClient.clientLogger.debug("Receiver run");
+        currentThread = Thread.currentThread();
+        connect();
     }
 
-    public void listen() {
+    /**
+     * first we check the information at  our input stream, if we haven't- we should check the EOS,
+     * in this case we break the loop and make some job for safely receiver closing
+     * if we have another case, without EOS , we will sleep for 10 mil. and check data available one more time
+     * After we get data to our input stream we should read string and send  string to console.
+     */
+    public void connect() {
+        /*
         try {
             System.out.println("receiver wait message");
             while (reader.hasNextLine()) {
 
                 String message = reader.nextLine();
-                System.out.println("get message to receiver "+message);
+                System.out.println("get message to receiver " + message);
                 System.out.println(message);
             }
         } finally {
-            closeReceiver();
+            System.out.println("finally connect");
+            disconnect();
         }
+*/
+
+        isOpen = true;
+        ChatClient.clientLogger.debug("Start waiting messages from server, receiver isOpen -{}",isOpen);
```

Comment:
Usually each class defines its own Logger, so here you should have 
private static final Logger LOG = LoggerFactory.get(Receiver.class);

and

LOG.debug("")

It provides an ability to switch off logging for particular class, or change logging level.